frontend/lib/Models/group_endpoint_provider.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:adpv_frontend/DataModels/group_endpoints.dart';
+import 'package:flutter/material.dart';
+
+import '../Repository/AdminRepository/groups_repository.dart';
+
+class GroupEndpointProvider with ChangeNotifier {
+  GroupEndpointsData groupEndpointsData = GroupEndpointsData.empty();
+  GroupsRepository repository = GroupsRepository();
+
+  GroupEndpointProvider(Future<GroupEndpointsData> future) {
+    init(future);
+  }
+
+  void init(Future<GroupEndpointsData> future) {
+    future.then((value) {
+      groupEndpointsData = value;
+      groupEndpointsData.sortFields();
+      notifyListeners();
+    });
+  }
+
+  void updateEndpointState(
+    bool? value,
+    String endpointLabel,
+  ) {
+    if (value != null) {
+      groupEndpointsData.endpoints[endpointLabel]!.isBelongingToGroup = value;
+    }
+    notifyListeners();
+  }
+
+  void updateFieldState(
+    bool? value,
+    String endpointLabel,
+    String fieldLabel,
+  ) {
+    if (value != null) {
+      groupEndpointsData.endpoints[endpointLabel]!.fields[fieldLabel]!
+          .isBelongingToGroup = value;
+    }
+    notifyListeners();
+  }
+
+  Future<GroupEndpointsData> save() =>
+      repository.updateGroupEndpoints(groupEndpointsData).then((value) async {
+        if (value == true) {
+          groupEndpointsData =
+              await repository.getEndpointsForGroup(groupEndpointsData.groupId);
+          groupEndpointsData.sortFields();
+          notifyListeners();
+        }
+        return groupEndpointsData;
+      });
```

Comment:
Po otrzymaniu zwrotki po update'cie nie ma sensu jeszcze raz wywoływać `getEndpointsForGroup`, ponieważ `PUT` i `GET` zwracają te same dane. Zerknij na swagger'a

Reply:
my bad

Reply:
otóż nie, nie zwracają tego samego

Reply:
zostaje tak, dopóki nie zwrócą tych samych jsonow, są różnice

Reply:
A w jakich polach?

Reply:
Backend został ujednolicony, można to poprawić